aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/Culture.cs

```diff
@@ -0,0 +1,38 @@
+using System.Globalization;
+
+namespace BindTryParseAPI.Models
+{
+    // <snippet>
+    public class Culture
+    {
+        public string? DisplayName { get; }
+
+        public Culture(string displayName)
+        {
+            DisplayName = displayName;
+        }
+
+        public static bool TryParse(string? value, out Culture? culture)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                culture = default;
+                return false;
+            }
+
+            try
+            {
+                CultureInfo ci = new CultureInfo(value);
+            }
+            catch
+            {
+                culture = default;
+                return false;
+            }
```

Comment:
@fiyazbinhasan  I had to put in this hacky try catch to get `TryParse ` to work correctly.  With your latest code when you enter `12` for the culture
`/12/WeatherForecast/RangeWithCulture?range=07/12/2022-07/14/2022`
you get a run-time exception. With my hack code, I get a nice error message.
I think I'll remove all the culture code unless you can figure out a way to make it clean.

Reply:
@Rick-Anderson I suggested throwing out this Culture class altogether and using it as a string to demonstrate the second example of DateRange.

Reply:
I'm not sure why we would want to pass an integer like 12, where the culture display name should be something en-GB, en-US, etc.

`/en-US/WeatherForecast/RangeWithCulture?range=07/12/2022-07/14/2022`

Reply:
@fiyazbinhasan 
> I'm not sure why we would want to pass an integer like 12, where the culture display name should be something en-GB, en-US, etc.

But that's the entire point of `TryParse`, it returns false if it can't do the conversion. It doesn't have to be `12`, it can be any miss-typed string that can't be turned into a culture.

> @Rick-Anderson I suggested throwing out this Culture class altogether and using it as a string to demonstrate the second example of DateRange.

Good point. Maybe we should leave it out and let it throw when someone enters a culture that can't be converted. 
Maybe you or @brunolins16 can think of a clever way to `TryParse` culture

Reply:
@Rick-Anderson Oh got it. I thought you wanted to remove the culture class. In that case, the way you converted the string to culture using try/catch seems ok. I guess it's the only way you can do it. :)

Reply:
But the code could be cleaned up to replace 
```
 public string? DisplayName { get; }

        public Culture(string displayName)
        {
            DisplayName = displayName;
        }
```

with saving `new CultureInfo(value);` and not `DisplayName`
let me know if you can do that.

Reply:
@Rick-Anderson you can do something like this,

```
public class Culture
    {
        public CultureInfo? CultureInfo { get; }

        public Culture(CultureInfo cultureInfo)
        {
            CultureInfo = cultureInfo;
        }

        public static bool TryParse(string? value, out Culture? culture)
        {
            if (string.IsNullOrEmpty(value))
            {
                culture = default;
                return false;
            }

            try
            {
                var ci = new CultureInfo(value);
                culture = new Culture(ci);
                return true;
            }
            catch
            {
                culture = default;
                return false;
            }
        }
    }
```

Also, I figured out that some culture uses dash (-) as a separator symbol. In that case, it's better to go with a comma for a range separator. Let me know if you want me to create a pull request with those changes.

Reply:
> Also, I figured out that some culture uses dash (-) as a separator symbol. In that case, it's better to go with a comma for a range separator. Let me know if you want me to create a pull request with those changes.

Yes, please do. I'll `git pull origin main` to bring in your changes.

Reply:
@Rick-Anderson could you take a look at the branch, and see if the samples are enough. If I get a go I'll create a PR afterwards,

https://github.com/fiyazbinhasan/Docs/tree/try-parse-with-iparsable/aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse